src/main/java/org/opentripplanner/graph_builder/linking/SimpleStreetSplitter.java

```diff
@@ -73,18 +73,23 @@
  * See discussion in pull request #1922, follow up issue #1934, and the original issue calling for replacement of
  * the stop linker, #1305.
  */
-public class SimpleStreetSplitter {
+public class SimpleStreetSplitter implements StreetSplitter {
 
     private static final Logger LOG = LoggerFactory.getLogger(SimpleStreetSplitter.class);
 
     public static final int MAX_SEARCH_RADIUS_METERS = 1000;
 
     private Boolean addExtraEdgesToAreas = false;
 
-    private StreetEdgeFactory edgeFactory;
+    private StreetEdgeFactory edgeFactory = new DefaultStreetEdgeFactory();
 
     public static final int WARNING_DISTANCE_METERS = 20;
 
+    // DESTRUCTIVE_SPLIT means edges are split and new edges are created (used when linking transit stops etc. during graph building)
+    // NON_DESTRUCTIVE_SPLIT means new temporary edges are created and no edges are deleted (Used when searching for origin/destination)
+    public static final boolean DESTRUCTIVE_SPLIT = true;
```

Comment:
Increasingly, convention is to avoid the use of constants to provide untyped pseudo-enum behavior. The concern is that these constants can be used, possibly accidentally, anywhere a boolean is expected, e.g. a method parameter called `boolean nonDestructive` (giving `nonDestructive = DESTRUCTIVE_SPLIT` which is actually non-destructive). That said, I concede that it is very readable. But I already find `destructive = true` and `destructive = false` to be quite clear in meaning. If we really wanted to use symbolic names here I think the normal convention would be an enum. But an enum with two mutually exclusive values is just a boolean.

Reply:
I have used this type of constants in the past, it is very informative when they are used as method arguments like: `edge.split(DESTRUCTIVE_SPLIT)`. So, I tend to prefer using them -even if they are not needed for clearity in  a variable assignment.

Reply:
Good to have some more input on these style subjects. It's true the constants make the author's intention very clear. The main (potential) problem is just that they can accidentally lead to deceptively "true" statements if someone later uses them outside their intended context.